cellengine/resources/attachment.py

```diff
@@ -1,25 +1,42 @@
 from __future__ import annotations
-from dataclasses import dataclass, field
-from dataclasses_json.cfg import config
+from typing import Optional
+
+from attr import define, field
 
 import cellengine as ce
-from cellengine.utils.dataclass_mixin import DataClassMixin, ReadOnly
+from cellengine.utils import converter, readonly
+
 
+try:
+    from typing import Literal
+except ImportError:
+    from typing_extensions import Literal
 
-@dataclass
-class Attachment(DataClassMixin):
+
+@define
+class Attachment:
     """A class representing a CellEngine attachment.
     Attachments are non-data files that are stored in an experiment.
     """
 
+    _id: str = field(on_setattr=readonly)
     filename: str
-    _id: str = field(
-        metadata=config(field_name="_id"), default=ReadOnly()
-    )  # type: ignore
-    crc32c: str = field(repr=False, default=ReadOnly())  # type: ignore
-    experiment_id: str = field(repr=False, default=ReadOnly())  # type: ignore
-    md5: str = field(repr=False, default=ReadOnly())  # type: ignore
-    size: int = field(repr=False, default=ReadOnly())  # type: ignore
+    crc32c: str = field(on_setattr=readonly, repr=False)
+    experiment_id: str = field(on_setattr=readonly, repr=False)
+    md5: str = field(on_setattr=readonly, repr=False)
+    size: int = field(on_setattr=readonly, repr=False)
+    grid_id: Optional[str] = None
+    type: Optional[Literal["textbox", "image"]] = None
+
+    @property
+    def client(self):
+        return ce.APIClient()
```

Comment:
Comment from earlier PR: I don't think resources should have a `client` property. Internally I'd just call `ce.APIClient` where you need to use it. It increases the API surface without any value that I see.

(https://github.com/primitybio/cellengine-python-toolkit/pull/112#discussion_r758884826)

Reply:
:+1: